Sources/SentryCrash/Recording/SentryCrashBinaryImageCache.c

```diff
@@ -0,0 +1,156 @@
+#include "SentryCrashBinaryImageCache.h"
+#include "SentryCrashDynamicLinker.h"
+#include <mach-o/dyld.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#if TEST || TESTCI
```

Comment:
`m`: I'm not a big fan of these compiler directives, as we are not testing the else branch. I would prefer to restructure the code so we don't need the compiler directives. Then we can test all the code.

Reply:
I understand your concern.

You are worried about someone accidentally changing the 'else' part. However, the code is so specific that I can't see that happening without breaking the build, unless someone does something malicious, which should be caught during the PR review.

IMO this is the most elegant way to add a "Wrapper" for C code without the need to compile extra code that wont be required for release.

And we do have a test that uses the native function just to make sure it dont break. https://github.com/getsentry/sentry-cocoa/blob/9b2cb171e12b0cc4efe48a38b602e2dd6bf3e5bf/Tests/SentryTests/Integrations/SentryCrash/SentryCrashIntegrationTests.swift#L179-L208

Reply:
That works for me 👍